To: vim_dev@googlegroups.com
Subject: Patch 7.3.243
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.243
Problem:    Illegal memory access in readline().
Solution:   Swap the conditions. (Dominique Pelle)
Files:	    src/eval.c


*** ../vim-7.3.242/src/eval.c	2011-07-07 16:20:45.000000000 +0200
--- src/eval.c	2011-07-07 17:32:16.000000000 +0200
***************
*** 14318,14324 ****
  	tolist = 0;
  	for ( ; filtd < buflen || readlen <= 0; ++filtd)
  	{
! 	    if (buf[filtd] == '\n' || readlen <= 0)
  	    {
  		/* In binary mode add an empty list item when the last
  		 * non-empty line ends in a '\n'. */
--- 14318,14324 ----
  	tolist = 0;
  	for ( ; filtd < buflen || readlen <= 0; ++filtd)
  	{
! 	    if (readlen <= 0 || buf[filtd] == '\n')
  	    {
  		/* In binary mode add an empty list item when the last
  		 * non-empty line ends in a '\n'. */
*** ../vim-7.3.242/src/version.c	2011-07-07 17:15:29.000000000 +0200
--- src/version.c	2011-07-07 17:32:30.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     243,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
261. You find diskettes in your pockets when doing laundry.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
